tests/LuaThreadStress/LuaThreadStress.cpp

```diff
@@ -0,0 +1,154 @@
+// LuaThreadStress.cpp
+
+// Implements a stress-test of cLuaState under several threads
+
+#include "Globals.h"
+#include "Bindings/LuaState.h"
+#include <thread>
+#include <random>
+
+
+
+
+
+/** How long the threading test should run. */
+static const int NUM_SECONDS_TO_TEST = 10;
+
+/** (Shared) return value for all the threads.
+Only failures write to this variable, so if any thread fails, this value will contain a nonzero. */
+static std::atomic<int> g_Return(0);
+
+
+
+
+
+/** Retrieves a callback from the Lua state that can be later called.
+Calls the Lua function getCallback with a_Seed param to retrieve the callback. */
+static cLuaState::cCallbackPtr getCallback(cLuaState & a_LuaState, unsigned a_Seed)
+{
+	cLuaState::cLock lock(a_LuaState);
+	cLuaState::cCallbackPtr res;
+	a_LuaState.Call("getCallback", a_Seed, cLuaState::Return, res);
+	return res;
+}
+
+
+
+
+
+/** Runs a single thread that stress-tests the cLuaState object.
+a_LuaState is the Lua state on which to operate.
+a_Seed is the seed for the random number generator for this thread.
+a_ShouldTerminate is a bool flag that another thread sets to ask this thread to terminate. */
+static void runStress(cLuaState * a_LuaState, unsigned a_Seed, std::atomic<bool> * a_ShouldTerminate)
+{
+	std::minstd_rand rnd;
+	rnd.seed(a_Seed);
+	auto callbackSeed = static_cast<unsigned>(rnd());
+	auto callback = getCallback(*a_LuaState, callbackSeed);
+	while (!a_ShouldTerminate->load())
+	{
+		// Pick a random operation on the Lua state and peform it:
+		switch (rnd() % 4)
+		{
+			case 0:
+			{
+				// Get a new callback:
+				callbackSeed = callbackSeed + 1;
+				callback = getCallback(*a_LuaState, callbackSeed);
```

Comment:
It doesn't strictly need to, but indeed *something* is missing - either a `break`, or a `// fallthrough` comment